src/utils/callbackSet.ts

```diff
@@ -0,0 +1,38 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+  export module Utils {
+    export class CallbackSet<CB extends Function> {
+      private _values: CB[];
+
+      constructor() {
+        this._values = [];
+      }
+
+      public add(value: CB) {
+        if (this._values.indexOf(value) === -1) {
+          this._values.push(value);
+        }
+        return this;
+      }
+
+      public remove(value: CB) {
+        var index = this._values.indexOf(value);
+        if (index !== -1) {
+          this._values.splice(index, 1);
+        }
+        return this;
+      }
+
+      public values() {
+        return this._values;
+      }
+
+      public callCallbacks(...args: any[]) {
+        // no fat-arrow notation to set "this" to current "this" context
```

Comment:
This is an extremely common trick that we use within our codebase, and I'm not sure if it makes sense to list this fact every time we use the "function" for `this` scenario.  That said, I'm not sure if it's intuitive enough to pick up.

How do we inform a reader that we are using `function` for the specific reason of achieving the right `this` context without using a comment? (Ideally we would use single-arrow... but that does not exist)